### PR TITLE
A type is `1`-acyclic if and only if it is `0`-connected

### PR DESCRIPTION
**Pull request by tomdjong**
In the process I also formalized:
- a type is `0`-acyclic if and only if it is inhabited, and
- a map is `0`-acyclic if and only if it is surjective.

Feedback on the group-theoretic assumption in `1-acyclic-types.lagda.md` is particularly welcome!

### CODE REVIEW

File: src/synthetic-homotopy-theory/1-acyclic-types.lagda.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+</details>
+
+## Idea
+
+We characterize the
+[`1`-acyclic types](synthetic-homotopy-theory.truncated-acyclic-types.md) as the
+[`0`-connected types](foundation.0-connected-types.md).
+
+In one direction, our proof relies on the following group-theoretic fact: the
+map of generators from a set `X` to the free group on `X` is injective. This is
```
Comment (fredrik-bakke) [inline src/synthetic-homotopy-theory/1-acyclic-types.lagda.md:42]:
```suggestion
map of [generators](group-theory.generating-elements-groups.md) from a [set](foundation-core.sets.md) `X` to the free group on `X` is [injective](foundation-core.injective-maps.md). This is
```

File: src/synthetic-homotopy-theory/1-acyclic-types.lagda.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+proved constructively in \[MRR88\] by Mines, Richman and Ruitenburg, and carried
+out in HoTT/UF and formalized in Agda in \[BCDE21\] by Bezem, Coquand, Dybjer
+and Escardó.
+
+Translated to [concrete groups](group-theory.concrete-groups.md) this means that
+for every set `X`, we have a pointed `1`-type `pt : BG` together with an
```
Comment (fredrik-bakke) [inline src/synthetic-homotopy-theory/1-acyclic-types.lagda.md:48]:
```suggestion
for every set `X`, we have a [pointed](structured-types.pointed-types.md) [`1`-type](foundation-core.1-types.md) `pt : BG` together with an
```

File: src/synthetic-homotopy-theory/1-acyclic-types.lagda.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+## See also
+
+- [Acyclic maps](synthetic-homotopy-theory.acyclic-maps.md)
+- [Acyclic types](synthetic-homotopy-theory.acyclic-types.md)
+- [`k`-acyclic types](synthetic-homotopy-theory.truncated-acyclic-types.md)
+- [`k`-acyclic maps](synthetic-homotopy-theory.truncated-acyclic-maps.md)
+- [Dependent epimorphisms](foundation.dependent-epimorphisms.md)
+- [Epimorphisms](foundation.epimorphisms.md)
+- [Epimorphisms with respect to sets](foundation.epimorphisms-with-respect-to-sets.md)
+- [Epimorphisms with respect to truncated types](foundation.epimorphisms-with-respect-to-truncated-types.md)
```
Comment (fredrik-bakke) [inline src/synthetic-homotopy-theory/1-acyclic-types.lagda.md:171]: Same here

File: src/synthetic-homotopy-theory/truncated-acyclic-types.lagda.md [hunk 4]
```diff
@@ -96,6 +96,19 @@ module _
...
+
+```agda
+module _
+  {l : Level} {k : 𝕋} {A : UU l}
+  where
+
+  is-truncated-acyclic-is-truncated-succ-acyclic :
+    is-truncated-acyclic (succ-𝕋 k) A → is-truncated-acyclic k A
+  is-truncated-acyclic-is-truncated-succ-acyclic =
+    is-connected-is-connected-succ-𝕋 k
```
Comment (fredrik-bakke) [inline src/synthetic-homotopy-theory/truncated-acyclic-types.lagda.md:109]: The name `is-truncated-acylic-is-truncated-acyclic-succ` more closely resembles the type signature

**Comment by tomdjong** [comment]
@fredrik-bakke I owe you a drink or two at the next conference we both attend :slightly_smiling_face:

File: src/synthetic-homotopy-theory/1-acyclic-types.lagda.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+open import synthetic-homotopy-theory.truncated-acyclic-types
+```
+
+</details>
+
+## Idea
+
+A type is [`1`-acyclic](synthetic-homotopy-theory.truncated-acyclic-types.md) if
+its [suspension](synthetic-homotopy-theory.suspensions-of-types.md) is
+[`1`-connected](foundation.connected-types.md)
```
Comment (fredrik-bakke) [inline src/synthetic-homotopy-theory/1-acyclic-types.lagda.md:39]:
```suggestion
[`1`-connected](foundation.connected-types.md).
```

File: src/synthetic-homotopy-theory/0-acyclic-maps.lagda.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+open import synthetic-homotopy-theory.truncated-acyclic-maps
+```
+
+</details>
+
+## Idea
+
+A [`0`-acyclic map](synthetic-homotopy-theory.truncated-acyclic-maps.md) is a
+map whose fibers are
```
Comment (fredrik-bakke) [inline src/synthetic-homotopy-theory/0-acyclic-maps.lagda.md:24]:
```suggestion
map whose [fibers](foundation-core.fibers-of-maps-md) are
```

File: src/synthetic-homotopy-theory/1-acyclic-types.lagda.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+open import synthetic-homotopy-theory.loop-spaces
+open import synthetic-homotopy-theory.truncated-acyclic-maps
+open import synthetic-homotopy-theory.truncated-acyclic-types
+```
+
+</details>
+
+## Idea
+
+A type is [`1`-acyclic](synthetic-homotopy-theory.truncated-acyclic-types.md) if
```
Comment (fredrik-bakke) [inline src/synthetic-homotopy-theory/1-acyclic-types.lagda.md:37]: Let's see, this link seems misplaced. Here you're defining `1`-acyclic, so I would write it in bold face. Moreover, you're not actually referring to `n`-truncated acyclic maps for general `n`, so I would not link to it. I.e.

```suggestion
A type is **`1`-acyclic** if
```

**Comment by fredrik-bakke** [comment]
Looking forward to it! I owe you one in return for writing such excellent PRs 😄

File: src/synthetic-homotopy-theory/0-acyclic-types.lagda.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+open import synthetic-homotopy-theory.0-acyclic-maps
+open import synthetic-homotopy-theory.truncated-acyclic-maps
+open import synthetic-homotopy-theory.truncated-acyclic-types
+```
+
+</details>
+
+## Idea
+
+A type is [`0`-acyclic](synthetic-homotopy-theory.truncated-acyclic-types.md) if
```
Comment (fredrik-bakke) [inline src/synthetic-homotopy-theory/0-acyclic-types.lagda.md:29]: Same as previous comment
```suggestion
A type is **`0`-acyclic** if
```

**Review by fredrik-bakke — APPROVED** [review]
Excellent work! I will merge this PR once my final comments are resolved :)

File: src/synthetic-homotopy-theory/1-acyclic-types.lagda.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+- \[MRR88\]: Ray Mines, Fred Richman and Wim Ruitenburg. A Course in
+  Constructive Algebra. Universitext. Springer, 1988. doi:
+  10.1007/978-1-4419-8640-5.
+
+- \[Wär23\]: David Wärn. Path spaces of pushouts. Preprint.
+  https://dwarn.se/po-paths.pdf. 2023.
+
+## See also
+
+- [`k`-acyclic maps](synthetic-homotopy-theory.truncated-acyclic-maps.md)
```
Comment (fredrik-bakke) [inline src/synthetic-homotopy-theory/1-acyclic-types.lagda.md:169]: DId you mean to link `k`-acyclic types?

```suggestion
- [`k`-acyclic types](synthetic-homotopy-theory.truncated-acyclic-types.md)
```

File: src/synthetic-homotopy-theory/1-acyclic-types.lagda.md (hunk 7 above)
Comment (tomdjong) [reply src/synthetic-homotopy-theory/1-acyclic-types.lagda.md:37]: Ah, the thinking was that `1`-acyclic is just a special case of `k`-acyclic of course. So it's not really a new definition. But I see your point and will implement it and fix the _See also_ links.

**Comment by tomdjong** [comment]
Just need to push.

File: src/synthetic-homotopy-theory/0-acyclic-types.lagda.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+open import synthetic-homotopy-theory.truncated-acyclic-types
+```
+
+</details>
+
+## Idea
+
+A type is **`0`-acyclic** if its
+[suspension](synthetic-homotopy-theory.suspensions-of-types.md) is
+[`0`-connected](foundation.0-connected-types.md)
```
[the remark this answers is not in the document]
Comment (tomdjong) [reply src/synthetic-homotopy-theory/0-acyclic-types.lagda.md:31]: Sorry, mixed GitHub suggestions and local edits too much :upside_down_face:

**Comment by tomdjong** [comment]
Really sorry about disabling auto-merge again :joy:

**Comment by fredrik-bakke** [comment]
No worries!

**Pull request merged**